lib/routes/Character/hooks/useCharacter.tsx

```diff
@@ -41,15 +41,16 @@ export function useCharacter(characterFromProps?: ICharacter | undefined) {
     }
   }, [characterFromProps]);
 
-  function loadTemplate(type: CharacterTemplates) {
+  async function loadTemplate(type: CharacterTemplates) {
+    const defaultCharacter = await CharacterFactory.make(type);
```

Comment:
needed to move this up because I didnt want to use the await operator inside an immer producer